deploy.yaml

```diff
@@ -0,0 +1,1164 @@
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: cainjector
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/component: "cainjector"
+  template:
+    metadata:
+      labels:
+        app: cainjector
+        app.kubernetes.io/name: cainjector
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: "cainjector"
+        app.kubernetes.io/version: "v1.7.1"
+    spec:
+      serviceAccountName: cert-manager-cainjector
+      securityContext:
+        runAsNonRoot: true
+      volumes:
+        - name: kubeconfig
+          configMap:
+            name: kubeconfig
+      containers:
+        - name: cert-manager
+          image: "docker.io/fabianvf/cert-manager-cainjector:v1.7.0-beta.0.150-f554a48dfa678a-dirty"
+          imagePullPolicy: IfNotPresent
+          args:
+          - --v=2
+          - --leader-election-namespace=kube-system
+          volumeMounts:
+            - name: kubeconfig
+              mountPath: /kubeconfig
+          env:
+          - name: KUBECONFIG
+            value: /kubeconfig
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+---
+# Source: cert-manager/templates/deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cert-manager
+  namespace: "cert-manager"
+  labels:
+    app: cert-manager
+    app.kubernetes.io/name: cert-manager
+    app.kubernetes.io/instance: cert-manager
+    app.kubernetes.io/component: "controller"
+    app.kubernetes.io/version: "v1.7.1"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: cert-manager
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/component: "controller"
+  template:
+    metadata:
+      labels:
+        app: cert-manager
+        app.kubernetes.io/name: cert-manager
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: "controller"
+        app.kubernetes.io/version: "v1.7.1"
+      annotations:
+        prometheus.io/path: "/metrics"
+        prometheus.io/scrape: 'true'
+        prometheus.io/port: '9402'
+    spec:
+      serviceAccountName: cert-manager
+      securityContext:
+
+        runAsNonRoot: true
+      volumes:
+        - name: kubeconfig
+          configMap:
+            name: kubeconfig
+      containers:
+        - name: cert-manager
+          image: "docker.io/fabianvf/cert-manager-controller:v1.7.0-beta.0.150-f554a48dfa678a-dirty"
+          imagePullPolicy: IfNotPresent
+          volumeMounts:
+          - name: kubeconfig
+            mountPath: /kubeconfig
+          args:
+          - --v=2
+          - --cluster-resource-namespace=$(POD_NAMESPACE)
+          - --leader-election-namespace=kube-system
+          ports:
+          - containerPort: 9402
+            protocol: TCP
+          env:
+          - name: KUBECONFIG
+            value: /kubeconfig
```

Comment:
this probably should point to the key from configMap. While running against a kind cluster, this worked locally:
```yaml
          env:
          - name: KUBECONFIG
            valueFrom:
              configMapKeyRef:
                name: kubeconfig
                key: kubeconfig
```